### PR TITLE
Capacitor 5 update

### PR DESCRIPTION
**Pull request by MateiNenciu**
This PR updates this plugin so it can be used with Capacitor 5. All steps from: https://capacitorjs.com/docs/updating/plugins/5-0 were followed + some updates to support BuildConfig from gradle 8+

This update does not modify minSdkVersion so it should work for clients who are switching to capacitor v5. Any suggestions are welcomed, maybe switching to a greater version like 7 (because this targets Java 17 and gradle 8+)

@pazlavi

### CODE REVIEW

**Comment by parthnayak7448** [comment]
Waiting for this PR to merge so I can integrate in my project using Cap 5

**Comment by ppriyankasen** [comment]
Same here, waiting for this PR to get merged to support cap5

**Comment by gelinger777** [comment]
+1

**Comment by MattijsE** [comment]
For people not wanting to wait: you can do `npm i https://github.com/nowjobs/appsflyer-capacitor-plugin/tree/main` to get capacitor 5 support inside your projects.

Once Appsflyer releases the Capacitor 5 version, then this fork will be removed.

**Comment by MateiNenciu** [comment]
@MattijsE I'm curious, why did you added the dist folder?

**Comment by MattijsE** [comment]
@MateiNenciu When using your branch I couldn't get it to compile without an extra build step, since it couldn't find the definitions. So by adding the dist folder this problem is solved.

**Comment by sultanmyrza** [comment]
@MattijsE adding dist folder can be avoided.
In `package.json` add `"prepare": "npm run build"`. I use `prepare` in my [plugins](https://github.com/sultanmyrza/preview-camera/blob/3f214e4e8bc68d53ea77bd53b0c218a46bf91856/package.json#L45) when I want to install directly from github without publishing to npm etc.

**Comment by ppriyankasen** [comment]
@MattijsE Is there any target date to merge this PR?

**Comment by MateiNenciu** [comment]
The target date is when the maintainer decides to merge my PR or do the changes itself

**Review by gelinger777 — APPROVED** [review]
seems it's ok

**Review by parthnayak7448 — APPROVED** [review]

**Comment by parthnayak7448** [comment]
@pazlavi can you provide us update on this PR please

**Comment by pazlavi** [comment]
Updating the plugin and releasing a new version is on our roadmap for Q3. I don't have an ETA at this moment.

**Comment by MateiNenciu** [comment]
So basically you are saying that we should wait for Q3 to merge a simple platform update?

**Comment by pazlavi** [comment]
We'll not release any new version without running a proper QA cycle and documenting the breaking changes. At this moment, our resources focus on other tasks.

**Comment by MateiNenciu** [comment]
Ok, anyone who want to use Capacitor v5 please feel free to use my fork till this understaffed corporation will handle this. Capacitor v5 was announced from april, it's not like it was released 2 days ago.
  I will strongly advise my data dept to look for an alternative MMP

**Review by suyundukov — APPROVED** [review]

**Comment by louison** [comment]
I effectively used the fork from @MateiNenciu in my current project and everything is working well. @suyundukov @gelinger777 @parthnayak7448 could you please find someone to approve and merge this PR once and for all 😄 ?
Thank you in advance !

**Comment by pazlavi** [comment]
`6.12.1`  release and support now Capacitor 5

**Pull request closed without merging**
